src/data/slides.tsx

```diff
@@ -0,0 +1,261 @@
+import * as dmv from 'dicom-microscopy-viewer'
+
+import { InstancesMetadata } from '../utils/types'
+
+/**
+ * Slide state
+ * @params frameofReferenceUID - array of series states
+ * @params containerIdentifier - array of series states
+ * @params areImagesMonochrome - type of images
+ * @params isMultiplexedSamples - is multi channel datasets
+ * @params key - key series connected to the slide
+ * @params seriesUIDsList - array of series UIDs connected to the slide
+ * @params keyOpticalPathIdentifier - key optical path indentifier
+ * @params opticalPathIdentifiersList - array of all optical path identifiers
+ * @params description - slide description, i.e.,
+ *                       Multiplexed-Samples, Monochrome Slide, RGB Slide
+ * @params volumeMetadata - array of volume metadata
+ * @params labelMetadata - array of label metadata
+ * @params overviewMetadata - array of overview metadata
+ */
+class Slide {
+  frameofReferenceUID: string = ''
+  containerIdentifier: string = ''
+  areImagesMonochrome: boolean = false
+  isMultiplexedSamples: boolean = false
+  key: string = ''
+  seriesUIDsList: string[] = []
+  keyOpticalPathIdentifier: string = ''
+  opticalPathIdentifiersList: string[] = []
+  description: string = ''
+  volumeMetadata: object[] = []
+  labelMetadata: object[] = []
+  overviewMetadata: object[] = []
+}
+
+/**
+ * Transforms series states array into a slide states array
+ * A series state has 3 array of metadata (volume, label
+ * and overview) already donwloaded.
+ *
+ * First we group in slides by FrameofReferenceUID and ContainerIdentifier:
+ * i.e. putting togheter images (overview, label, volume) instances from N series.
+ *
+ * Secondly we indentify the slides as:
+ *   A) If the number of opticalPathIdentifier > 1 and SamplesPerPixel === 1
+ *      and PhotometricInterpretation === MONOCHROME2, then the observation is a
+ *      multiplexed samples with N "channels";
+ *   B) If the number of opticalPathIdentifier === 1 and SamplesPerPixel === 1
+ *      and PhotometricInterpretation === MONOCHROME2, then the observation is a
+ *      simple single monochorme image sample;
+ *   C) If the number of opticalPathIdentifier === 1 and SamplesPerPixel !== 1
+ *      and PhotometricInterpretation === RGB or YBR_*,
+ *      then the observation is a RGB single image sample.
+ *
+ * @params instancesMetadataArray - array of instances from series, each element
+ *         of the array corresponds to a series
+ * @params initiallySelectedSeriesInstanceUID - to visualize
+ *         at first loading data coming from a specific series.
+ * @returns slides - array of slide states
+ */
+function createSlides (
+  instancesMetadataArray: InstancesMetadata[],
+  initiallySelectedSeriesInstanceUID: string = ''
```

Comment:
Please use `undefined` instead of a `''`:

```suggestion
  initiallySelectedSeriesInstanceUID?: string
```

Reply:
done in https://github.com/MGHComputationalPathology/slim/pull/25/commits/b457fc27cee8440c76bb315a09b78f4b05a20732